### PR TITLE
Fix jexl logging and update VariantFiltration doc.

### PR DESCRIPTION
**Pull request by cmnbroad**
Fixes https://github.com/broadinstitute/gatk/issues/5362 and https://github.com/broadinstitute/gatk/issues/4637. Log messages from JEXL were being dropped because of https://github.com/broadinstitute/gatk/issues/4637, making the behavior described by https://github.com/broadinstitute/gatk/issues/5362 even more subtle to discern. Now, instead of seeing:

> log4j:WARN No appenders could be found for logger (org.apache.commons.jexl2.JexlEngine).
> log4j:WARN Please initialize the log4j system properly.
> log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.

the user will see:

> 11:51:06.321 WARN  JexlEngine - ![51,60]: 'QD < 2.0 || MQ < 40.0 || FS > 60.0 || SOR > 3.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0 || ExcessHet > 54.69;' undefined variable MQRankSum

### CODE REVIEW

**Comment by sooheelee** [comment]
Here are the documents we should also update to reflect these updates:
- https://gatkforums.broadinstitute.org/gatk/discussion/1255/using-jexl-to-apply-hard-filters-or-select-variants-based-on-annotation-values
- https://software.broadinstitute.org/gatk/documentation/article?id=11080

**Comment by sooheelee** [comment]
Tagging @rcmajovski, who will lead on the forum doc updates.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/filters/VariantFiltration.java [hunk 1]
```diff
@@ -50,13 +50,52 @@
...
+ *    gatk VariantFiltration \
+ *    -R reference.fasta \
+ *    -V input.vcf.gz \
+ *    -O output.vcf.gz \
+ *    --filter-name "my_filter" \
+ *    --filter-expression "AB < 0.2 || MQ0 > 50"
+ *  </pre>
+ *
+ * <p>
+ * </p>Please see <a href='https://software.broadinstitute.org/gatk/documentation/article.php?id=1255'>this
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/filters/VariantFiltration.java]: This article is recommending the old way of filtering, are there plans to update it as part of this?

**Comment by cmnbroad** [comment]
@rcmajovski @sooheelee I'm tempted to take this link out of the gatk doc until there is an updated version of it online. Is there an eta for that, or a ticket somewhere, so I can make a ticket to restore the link in the tools that use JEXL when thats finished ?

**Comment by sooheelee** [comment]
@cmnbroad, sorry for the delayed response. I was in Taiwan giving a workshop last week and then I fell ill (run of the mill cold virus).

The Comms team has been migrating issue tracking to a new system on Monday.com, which I am just now familiarizing myself to as I have been occupied with Taiwan workshop preparation. Forum questions are tracked in a separate system, Zendesk. The SOPs towards handling work for the two different systems are still under development so the best way to ensure you are up to date with the progress of work is to contact Robert @rcmajovski. The previous GitHub board that we used at https://github.com/broadinstitute/dsde-docs is still up and I still have my issue tickets here as I haven't had a chance to migrate these. 

I do not know if you have access, but here is the link to track the issue ticket on Monday.com:
https://dsp-comms.monday.com/boards/145112271/

Just to orient you, if you click on the issue, a sideboard slides out from the right and you can comment on the work:
screenshot 2018-12-13 22 51 00

It seems I've been assigned to update these documents. I'm unfamiliar with JEXL itself. I will survey the work that needs to be done and let you know.

**Comment by sooheelee** [comment]
The documentation clarity issue is resolved with the publication of a new forum tutorial at https://software.broadinstitute.org/gatk/documentation/article?id=23216. 
screenshot 2019-02-01 17 48 19

**Pull request merged**